### PR TITLE
test: convert sqs.test.js over to use localstack for end-to-end tests (similar to s3.test.js)

### PR DESCRIPTION
**Pull request by trentm**
- update to the latest localstack
- restore testing of recent releases of aws-sdk (skipping the "bad" 2.1372.0 release that attempted to switch SQS over to the new "JSON protocol")
- changes the SQS "integration tests" to run use-sqs.js against localstack, instead of just against a local mock SQS server with canned responses

Fixes: #3324

### CODE REVIEW

**Review by v1v — COMMENTED** [review]
/test tav aws-sdk

**Comment by trentm** [comment]
With the upgrade to localstack 2.1.0 the raw response body from the S3 server
changed slightly, requiring an update to the expected `encoded_body_size` in s3.test.js.

```js
// old httpResponse.body
<ListAllMyBucketsResult xmlns="http://s3.amazonaws.com/doc/2006-03-01"><Owner><ID>bcaf1ffd86f41161ca5fb16fd081034f</ID><DisplayName>webfile</DisplayName></Owner><Buckets></Buckets></ListAllMyBucketsResult>

// new httpResponse.body:
<?xml version='1.0' encoding='utf-8'?>
<ListAllMyBucketsResult><Owner><DisplayName>webfile</DisplayName><ID>75aa57f09aa0c8caeab4f8c24e99d10f8e7faeebf76c078efc7c6caea54ba06a</ID></Owner><Buckets /></ListAllMyBucketsResult>
```

File: test/instrumentation/modules/@aws-sdk/client-s3.test.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ tape.test('simple S3 V3 usage scenario', function (t) {
               cloud: { region: 'us-east-2' },
               service: { type: '', name: '', resource: 's3' }
             },
-            http: { status_code: 200 }
+            http: { status_code: 200, response: { encoded_body_size: 222 } }
```
Comment (trentm) [inline test/instrumentation/modules/@aws-sdk/client-s3.test.js:133]: Review note: Changes to `encoded_body_size` in these tests are due to the localstack update which slightly changed the response XML for S3. See https://github.com/elastic/apm-agent-nodejs/pull/3334#issuecomment-1590196688 for an example.

**Review by david-luna — APPROVED** [review]

**Review by trentm — COMMENTED** [review]
/test tav aws-sdk

**Comment by trentm** [comment]
> /test tav aws-sdk

@david-luna Needs to be done as a *review* comment to be picked up by tav-command.yml.

File: test/instrumentation/modules/aws-sdk/fixtures/use-sqs.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  params = {
+    QueueUrl: queueUrl,
+    MaxNumberOfMessages: 10,
+    AttributeNames: ['All'],
+    MessageAttributeNames: ['All'],
+    VisibilityTimeout: 10,
+    WaitTimeSeconds: 5
+  }
+  const messages = []
+  for await (const attemptNum of [0, 1, 2, 3, 4]) {
```
[the remark this answers is not in the document]
Comment (trentm) [reply test/instrumentation/modules/aws-sdk/fixtures/use-sqs.js]: commit 8d1fa57

**Comment by trentm** [comment]
> Needs to be done as a _review_ comment to be picked up by tav-command.yml.

But then that didn't work either because (from https://github.com/elastic/apm-agent-nodejs/actions/runs/5316140304): 
"Canceling since a higher priority waiting request for 'tav-command-refs/pull/3334/merge' exists"
Sigh. That the "tav-command" workflow gets considered at all for the other pull request activities (like pushing another commit) is annoying.

**Review by trentm — COMMENTED** [review]
/test tav aws-sdk

**Review by trentm — COMMENTED** [review]
/test tav aws-sdk

**Review by trentm — COMMENTED** [review]
/test tav aws-sdk

**Pull request merged**